src/core/stream/period/period_stream.ts

```diff
@@ -157,26 +162,36 @@ export default function PeriodStream({
       const newStream$ = clock$.pipe(
         take(1),
         mergeMap((tick) => {
+          const { audioTrackSwitchingMode } = options;
           const qSourceBuffer = createOrReuseQueuedSourceBuffer(sourceBuffersStore,
                                                                 bufferType,
                                                                 adaptation,
                                                                 options);
           const strategy = getAdaptationSwitchStrategy(qSourceBuffer,
                                                        period,
                                                        adaptation,
-                                                       tick);
+                                                       tick,
+                                                       audioTrackSwitchingMode);
           if (strategy.type === "needs-reload") {
             return observableOf(EVENTS.needsMediaSourceReload(period, tick));
           }
 
           const cleanBuffer$ = strategy.type === "clean-buffer" ?
             observableConcat(...strategy.value.map(({ start, end }) =>
                                qSourceBuffer.removeBuffer(start, end))
-                            ).pipe(ignoreElements()) :
-            EMPTY;
+                            ).pipe(switchMap((_, i) => {
+                              if (audioTrackSwitchingMode === "flush" &&
```

Comment:
You didn't check here that the current adaptation is an audio one.
Using `i` is not necessary (and a little ugly as we repeat the initial logic here, we change one of both parts and everything is broken), we can either concat or do other operations with the source observables to be sure that we're flushing after everything is done.

But more importantly, I don't get why you aren't just using `getAdaptationSwitchStrategy` to return the wanted strategy (it even has the flush thing in its return type) and then just use what that function returned here.

Reply:
Yes, u are right.

I did a new commit.